your-rents-core/src/test/java/com/yourrents/core/service/PropertyServiceTest.java

```diff
@@ -20,87 +20,97 @@
  * #L%
  */
 
-import com.yourrents.core.EnvironmentTest;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.times;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
 import com.yourrents.core.NotFoundException;
 import com.yourrents.core.dto.Property;
 import com.yourrents.core.test.TestConfig;
-import org.jooq.DSLContext;
-import org.jooq.Record;
-import org.jooq.Result;
+import com.yourrents.data.jooq.daos.PropertyDao;
+import com.yourrents.data.jooq.tables.records.PropertyRecord;
+
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
-import java.io.IOException;
-import java.io.InputStream;
-import java.nio.charset.StandardCharsets;
-import java.util.UUID;
-
-import static com.yourrents.data.jooq.Tables.PROPERTY;
-import static org.assertj.core.api.Assertions.assertThat;
-import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
-
-
 @ExtendWith(SpringExtension.class)
 @ContextConfiguration(classes = TestConfig.class)
 class PropertyServiceTest {
     private static final UUID EXISTING_UUID = UUID.fromString("a444c18e-01ad-4aef-9365-6f42752faa62");
     private static final UUID NOT_EXISTING_UUID = UUID.fromString("e5c1ec0b-95d4-4e9f-b5d1-12c55812dbe7");
 
     @Autowired
-    private DSLContext dsl;
+    private PropertyDao mockPropertyDao;
```

Comment:
I'd have preferred using _@MockBean_

Reply:
Non c'è spring-boot in core, e non intendo avercelo.

Reply:
Ops, sorry...commenting in Italian today...it's Sunday! ;)